### ISSUE TITLE
How do I add a role object to the appRoles property?

### ISSUE THREAD

**Issue by isaacchua** [Azure/azure-cli#6875]
I'm trying at add an additional ```appRole``` but this doesn't seem to work.

```
az ad app update --id 12345678-1234-4321-1234-1234567890ab --add appRoles @"
     {
       "allowedMemberTypes": [
           "User"
       ],
       "description": "Administrators Only",
       "displayName": "Admin",
       "id": "4f8f8640-f081-492d-97a0-caf24e9bc134",
       "isEnabled": true,
       "origin": "ServicePrincipal",
       "value": "Administrator"
   }
"@
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 921aa920-c25f-5227-aabb-49116da38fa7
* Version Independent ID: 9b5a53ad-edf8-4d89-f8e6-cde01b7866f6
* Content: [az ad app](https://docs.microsoft.com/en-us/cli/azure/ad/app?view=azure-cli-latest)
* Content Source: [src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-role/azure/cli/command_modules/role/_help.py)
* Service: **active-directory**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yugangw-msft** [comment]
Graph role support is not supported yet. This actually is missing across all SDKs, We are looking for resources to on-board this.

**Comment by mattdot** [comment]
I'm trying to do the same thing.  This command actually seems to succeed and modify the application, but fails with `argument of type 'NoneType' is not iterable` when reading the response data.

```
az ad app update --id 0ff68287-25f5-4f9a-89d7-4f387994e2e8 --add appRoles=@"{"allowedMemberTypes":["User"], "description": "User","displayName": "User","id": "18d14569-c3bd-439b-9a66-3a2aee01d14f","isEnabled": true,"value": null}"@ --debug
```
logs
```
msrest.http_logger : Request URL: 'https://graph.windows.net/eb61ad58-722d-4237-a534-5945a8f870b5/applications/c983e753-17dc-4eac-bfcb-91b499e2dc25?api-version=1.6'
msrest.http_logger : Request method: 'GET'
msrest.http_logger : Request headers:
msrest.http_logger :     'User-Agent': 'python/3.6.5 (Linux-4.4.0-17763-Microsoft-x86_64-with-debian-stretch-sid) requests/2.19.1 msrest/0.5.5 msrest_azure/0.5.0 azure-graphrbac/0.40.0 Azure-SDK-For-Python AZURECLI/2.0.46'
msrest.http_logger :     'Accept-Encoding': 'gzip, deflate'
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'Connection': 'keep-alive'
msrest.http_logger :     'Authorization': '*****'
msrest.http_logger :     'x-ms-client-request-id': '8b5a8c34-d1db-11e8-ab40-54e1adf38bd0'
msrest.http_logger :     'CommandName': 'ad app update'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger : Request body:
msrest.http_logger : None
msrest.http_logger : Response status: 200
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Content-Type': 'application/json; odata=minimalmetadata; streaming=true; charset=utf-8'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'Server': 'Microsoft-IIS/10.0'
msrest.http_logger :     'ocp-aad-diagnostics-server-name': 'wIQemZ+5vqMn85aXsgJ+yzAwtRuiwLPflul5H4QC6Xs='
msrest.http_logger :     'request-id': 'dbbc0b6c-6b1d-4ad4-b4b3-1a25fd3373d3'
msrest.http_logger :     'client-request-id': '8b5a8c34-d1db-11e8-ab40-54e1adf38bd0'
msrest.http_logger :     'x-ms-dirapi-data-contract-version': '1.6'
msrest.http_logger :     'ocp-aad-session-key': '8DpX_A0ey1Qa4TKMMRqhtl8DFPNDjdbrvVya9HypCZ6eDvlrxL1kjrkJ4OewFA2AZV4ajZ68fTcq-RCkj03q1f1MgBg9x6jUbTYUIchXUqjGXVMPEpGyW_JU5v6TwAgFJkODg2oyfMLviCK6G2FXrg.IM6H55KysYqfxHqki0vHnHV8yFXChro5AtLe3WD2iRk'
msrest.http_logger :     'DataServiceVersion': '3.0;'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'Access-Control-Allow-Origin': '*'
msrest.http_logger :     'X-AspNet-Version': '4.0.30319'
msrest.http_logger :     'X-Powered-By': 'ASP.NET'
msrest.http_logger :     'Duration': '356225'
msrest.http_logger :     'Date': 'Wed, 17 Oct 2018 07:09:15 GMT'
msrest.http_logger :     'Content-Length': '2278'
msrest.http_logger : Response content:
msrest.http_logger : b'{"odata.metadata":"https://graph.windows.net/eb61ad58-722d-4237-a534-5945a8f870b5/$metadata#directoryObjects/@Element","odata.type":"Microsoft.DirectoryServices.Application","objectType":"Application","objectId":"c983e753-17dc-4eac-bfcb-91b499e2dc25","deletionTimestamp":null,"acceptMappedClaims":null,"addIns":[],"appId":"0ff68287-25f5-4f9a-89d7-4f387994e2e8","appRoles":[{"allowedMemberTypes":["User"],"description":"msiam_access","displayName":"msiam_access","id":"b9632174-c057-4f7e-951b-be3adc52bfe6","isEnabled":true,"value":null},{"allowedMemberTypes":["User"],"description":"User","displayName":"User","id":"18d14569-c3bd-439b-9a66-3a2aee01d14f","isEnabled":true,"value":null}],"availableToOtherTenants":false,"displayName":"clitest","errorUrl":null,"groupMembershipClaims":null,"homepage":null,"identifierUris":["https://identifier.clitest.local","https://clitest.local"],"informationalUrls":{"termsOfService":null,"support":null,"privacy":null,"marketing":null},"isDeviceOnlyAuthSupported":null,"keyCredentials":[],"knownClientApplications":[],"logoutUrl":null,"logo@odata.mediaEditLink":"directoryObjects/c983e753-17dc-4eac-bfcb-91b499e2dc25/Microsoft.DirectoryServices.Application/logo","logoUrl":null,"mainLogo@odata.mediaEditLink":"directoryObjects/c983e753-17dc-4eac-bfcb-91b499e2dc25/Microsoft.DirectoryServices.Application/mainLogo","oauth2AllowIdTokenImplicitFlow":true,"oauth2AllowImplicitFlow":false,"oauth2AllowUrlPathMatching":false,"oauth2Permissions":[{"adminConsentDescription":"Allow the application to access clitest on behalf of the signed-in user.","adminConsentDisplayName":"Access clitest","id":"fdadc2ee-2e1a-4480-9dca-922b2db8a00c","isEnabled":true,"type":"User","userConsentDescription":"Allow the application to access clitest on your behalf.","userConsentDisplayName":"Access clitest","value":"user_impersonation"}],"oauth2RequirePostResponse":false,"optionalClaims":null,"orgRestrictions":[],"parentalControlSettings":{"countriesBlockedForMinors":[],"legalAgeGroupRule":"Allow"},"passwordCredentials":[],"publicClient":null,"publisherDomain":"csepcf.com","recordConsentConditions":null,"replyUrls":["https://reply.clitest.local"],"requiredResourceAccess":[],"samlMetadataUrl":null,"signInAudience":"AzureADMyOrg","tokenEncryptionKeyId":null}'
argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 197, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 366, in execute
    cmd.exception_handler(ex)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/role/commands.py", line 31, in graph_err_handler
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 343, in execute
    result = cmd(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 182, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/arm.py", line 496, in handler
    add_properties(instance, arg_values, force_string)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/arm.py", line 806, in add_properties
    list_to_add_to = _find_property(instance, list_attribute_path)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/arm.py", line 985, in _find_property
    instance = _update_instance(instance, part, path)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/arm.py", line 975, in _update_instance
    elif part in getattr(instance, 'additional_properties', {}):
TypeError: argument of type 'NoneType' is not iterable
telemetry.save : Save telemetry record of length 2475 in cache
telemetry.check : Returns Positive.
telemetry.main : Begin creating telemetry upload process.
telemetry.process : Creating upload process: "/opt/az/bin/python3 /opt/az/lib/python3.6/site-packages/azure/cli/telemetry/__init__.py /home/matt/.azure"
telemetry.process : Return from creating porcess
telemetry.main : Finish creating telemetry upload process.
```

**Comment by StefanKert** [comment]
@yugangw-msft any news on this? It seems like it is currently not possible to automate adding `appRoles` which is somehow unsatisfying. 

@mattdot is this workaround still working for you? I tried updating our manifest but it was not modified by executing this command.

**Comment by yugangw-msft** [comment]
Folks, I should get this support in by the middle of March

**Comment by stweedie** [comment]
support has landed in [2.0.59](https://github.com/MicrosoftDocs/azure-docs-cli/blob/master/docs-ref-conceptual/release-notes-azure-cli.md#role), thanks @yugangw-msft!

Is there anyway this can be added (with examples preferably) to the [MS docs?](https://docs.microsoft.com/en-us/cli/azure/ad/app?view=azure-cli-latest)
